### PR TITLE
New data_kbuff directive v2

### PR DESCRIPTION
**Pull request by ofuhrer**
This PR implements the possibility to serialize data inside a k-blocked region of code. It buffers data(:,:,k) for each k-iteration and serializes data(:,:,:) out once a field is completely filled.

- Minor cleanup and making consistent of existing code.
- New Fortran file utils_ppser_kbuff.f90 that implements the buffering functionality (currently for 3d sp, dp, int fields and only in write mode).
- New !$ser data_kbuff k_value=k k_size=nz name=field(:,:) that can be used as a drop-in replacement of !$ser data name=field(:,:,:) but within a k-loop.
- Update of documentation.
- Minor cleanup of pp_ser.py (should have no impact on any of the users)

It implements all changes requested by @havogt in #220. This is a new PR since the other one was made from master, which was a very bad idea.

### CODE REVIEW

File: src/serialbox-fortran/utils_ppser_kbuff.f90 [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+    WRITE(0,*) 'DEBUG create_kbuff: savepoint=', TRIM(savepoint%savepoint_name)
+    WRITE(0,*) 'DEBUG create_kbuff: fieldname=', TRIM(fieldname)
+    WRITE(0,*) 'DEBUG create_kbuff: field_type=', field_type
+    WRITE(0,*) 'DEBUG create_kbuff: dim_i,dim_j,dim_k=', dim_i, dim_j, dim_k
+  END IF
+
+  ! security check
+  IF (kbuff_id < 1 .OR. kbuff_id > max_kbuff) THEN
+    WRITE(0,*) 'ERROR in utils_ppser_kbuff: illegal kbuff_id encountered'
+    STOP
```
Comment (lxavier) [inline src/serialbox-fortran/utils_ppser_kbuff.f90]: Are you sure you want a stop in utility routines ? Alternative would be to return and give error code (but then code needs to be checked on the user side)

**Comment by lxavier** [comment]
looks good to me ! I did not look at what is wrong with the test

File: src/serialbox-fortran/utils_ppser_kbuff.f90 (hunk 1 above)
Comment (ofuhrer) [reply src/serialbox-fortran/utils_ppser_kbuff.f90]: Good point. Unfortunately, there's nobody to catch the error, and connecting pp_ser.py with COSMO's error handler is not the way to go. I looked at `m_serialize.f90` how this is handled and adopted the `STOP` strategy from there. I think there is not good alternative.

**Comment by ofuhrer** [comment]
Re the Travis test: Not sure what is happening there as well. @havogt Is this an issue on Travis CI? has master been building successfully all the time?
image

**Comment by havogt** [comment]
Seems to be a travis issue. I restarted the build and other combinations failed. Now running the master.

**Comment by havogt** [comment]
launch jenkins

File: src/serialbox-fortran/CMakeLists.txt [hunk 2]
```diff
@@ -10,19 +10,23 @@
...
     )
-  
-  set(SOURCES_FORTRAN utils_ppser.f90 )
+  set(SOURCES_FORTRAN utils_ppser.f90 utils_ppser_kbuff.f90)
+
   if (SERIALBOX_ENABLE_FTG)
     set(SOURCES_FORTRAN ${SOURCES_FORTRAN} m_ser_ftg.f90 m_ser_ftg_cmp.f90)
   endif()
+  
+  if(SERIALBOX_FORTRAN_SAVEPOINT_UNIQUE_ID)
```
[the remark this answers is not in the document]
Comment (ofuhrer) [reply src/serialbox-fortran/CMakeLists.txt]: Done.

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**